extensions/hibernate-orm/runtime/src/main/java/io/quarkus/hibernate/orm/runtime/PersistenceUnitsHolder.java

```diff
@@ -54,7 +54,7 @@ public static RecordedState getRecordedState(String persistenceUnitName) {
         if (persistenceUnitName == null) {
             key = NO_NAME_TOKEN;
         }
-        return persistenceUnits.recordedStates.get(key);
+        return persistenceUnits.recordedStates.remove(key);
```

Comment:
We should maybe rename the method to `popRecordedState` or something similar? So that consumers know what they are doing?

Reply:
ok good point.

Reply:
OK, then, if we can just adjust the method name in this commit, we can merge.